app/models/apple/episode.rb

```diff
@@ -439,11 +441,7 @@ def synced_with_apple?
     end
 
     def waiting_for_asset_state?
-      (podcast_delivery_files.length > 0 &&
-        podcast_delivery_files.all?(&:delivered?) &&
-        podcast_delivery_files.all?(&:processed?) &&
-        !podcast_delivery_files.all?(&:processed_errors?) &&
-        !audio_asset_state_finished?)
+      podcast_container.delivery_settled? && !audio_asset_state_finished?
```

Comment:
so now only looking the episode's `audio_asset_state` and at the podcast container (which manages / coordinates the deliveries).